src/libraries/Common/src/Interop/Linux/OpenLdap/Interop.Ldap.cs

```diff
@@ -67,6 +68,26 @@ internal static partial class Ldap
     {
         static Ldap()
         {
+            // Register callback that tries to load other libraries when the default library "libldap-2.4.so.2" not found
+            AssemblyLoadContext.Default.ResolvingUnmanagedDll += (assembly, ldapName) =>
+            {
+                IntPtr handle = IntPtr.Zero;
+
+                if (ldapName != Libraries.OpenLdap)
+                {
+                    return handle;
+                }
+
+                if (NativeLibrary.TryLoad("libldap-2.5.so.0", out handle) ||
```

Comment:
Since 2.5 is the more widely used in newer distros, I wonder if we should let 2.5 be the default (meaning the one we use in the Libraries.OpenLdap name) and then we use this event as a fallback for the legacy (and preview) versions.

Reply:
Sounds good